src/NLog/Targets/FileArchiveModes/FileArchiveModeBase.cs

```diff
@@ -188,8 +189,16 @@ private bool ShouldDeleteFile(DateAndSequenceArchive existingArchiveFile, int re
             if (maxArchiveDays > 0)
             {
                 var fileDateUtc = existingArchiveFile.Date.Date.ToUniversalTime();
-                if (fileDateUtc > MaxAgeArchiveFileDate && (NLog.Time.TimeSource.Current.Time.Date.ToUniversalTime() - fileDateUtc).TotalDays > maxArchiveDays)
-                    return true;
+                if (fileDateUtc > MaxAgeArchiveFileDate)
+                {
+                    var currentDateUtc = NLog.Time.TimeSource.Current.Time.Date.ToUniversalTime();
+                    var fileAgeDays = (currentDateUtc - fileDateUtc).TotalDays;
+                    if (fileAgeDays > maxArchiveDays)
+                    {
+                        InternalLogger.Debug("FileTarget: Detected old file in archive. FileName={0}, FileDate={1}, FileDateUtc={2}, CurrentDateUtc={3}, Age={4} days", existingArchiveFile.FileName, existingArchiveFile.Date.ToString("u"), fileDateUtc.ToString("u"), currentDateUtc.ToString("u"), Math.Round(fileAgeDays, 1));
```

Comment:
PS I think you also could do `{2:u}` instead of `ToString("u")`. 

Maybe a bit easier to read, but not very important.

Reply:
I prefer "u" or "o". But find "u" more readable.

Reply:
I mean,

instead of:

```c#
InternalLogger.Debug("Date={0},", existingArchiveFile.FileName, Date.ToString("u"))
```

A more reabable way would be:

```c#
InternalLogger.Debug("Date={0:u},", existingArchiveFile.FileName, Date)
```

:)

Reply:
Well it still performs boxing etc. But if you really want it then ok.